main_script.py

```diff
@@ -0,0 +1,113 @@
+                    and the last argument should be a string specifying the action to be performed.
+
+    Returns:
+        str or list: The result of the specified action on the input sequences.
+
+    """
+
+    action = args[-1].lower()
+    sequences = args[:-1]
+    results = []
+
+    action_list = {
+        "transcribe": transcribe,
+        "reverse": reverse,
+        "complement": complement,
+        "reverse_complement": reverse_complement,
+    }
+
+    for sequence in sequences:
+        if all(base in drd.DNA_LETTERS or base in drd.RNA_LETTERS for base in sequence):
+            if 'U' in sequence and 'T' in sequence:
+                raise ValueError("Invalid sequence: Contains both U and T")
+            elif action == 'transcribe' and not set(sequence).issubset(drd.DNA_LETTERS):
+                raise ValueError("Invalid DNA sequence for transcription")
+            result = action_list[action](sequence)
+        else:
+            raise ValueError(f"Invalid sequence for procedure: {action}")
+        results.append(result)
+
+    return results if len(results) > 1 else results[0]
+
+
+def main_protein_tools(*args: str):
+    """
+    Main function to perform various actions on protein sequences.
+
+    Args:
+    - *args: Variable number of arguments. The first n-1 arguments should be protein sequences,
+             and the last argument should be a string specifying the action to be performed.
+
+    Returns:
+    - The result of the specified action on the input protein sequences.
+
+    Raises:
+    - ValueError: If the specified action is not supported or if there is an error in the number of sequences.
+                  Also raised if the input sequences are not valid protein sequences.
+
+    Supported Actions:
+    - "get_pI": Calculate isoelectric points for each amino acid in the sequence.
+    - "calculate_aa_freq": Calculate the frequency of each amino acid in a protein sequence.
+    - "translate_protein_rna": Translate amino acid sequence to RNA, using random codons for each amino acid.
+    - "convert_to_3L_code": Convert one-letter amino acid sequence to three-letter coding.
+    - "protein_mass": Calculate the molecular weight of the protein sequence.
+    """
+
+    action = args[-1]
+    sequences = args[:-1]
+    action_list = {
+        "get_pI": get_pI,
+        "calculate_aa_freq": calculate_aa_freq,
+        "translate_protein_rna": translate_protein_rna,
+        "convert_to_3L_code": convert_to_3L_code,
+        "protein_mass": protein_mass,
+    }
+
+    if action not in action_list:
+        raise ValueError(f"No such action: {action}")
+
+    for sequence in sequences:
+        if not set(sequence).issubset(prd.AA_LETTERS):
+            raise ValueError(f"The sequence is not protein sequence: {sequence}")
+
+    result = action_list[action](*sequences)
+
+    return result
+
+def main_fastq(input_path: str, output_filename: str = None):
```

Comment:
```suggestion
def run_fastq_filter(input_path: str, output_filename: str = None):
```

или

```suggestion
def filter_fastqinput_path: str, output_filename: str = None):
```


Тем не менее, тут большая ключевая ошибка. Надо было в исходной функции заменить прием словаря на прием пути. Ты добавила по сути "функцию-обертку" снаружи (и это в цело ок), но которая не принимает никаких аргументов для фильтрации... Как я могу воспользоваться фильтром? В этой функции я не могу задать условия для фильтра, в функции из дз 5 я не могу передать файл.